inspirehep/modules/orcid/utils.py

```diff
@@ -236,53 +78,39 @@ def _get_api_url_for_recid(server_name, api_endpoint, recid):
     return urljoin(api_url, recid)
 
 
-def _get_record_by_mime(config, recid, mime_type):
-    """
+def get_orcid_recid_key(orcid, rec_id):
+    """Return the string 'orcid:``orcid_value``:``rec_id``'"""
+    return 'orcid:{}:{}'.format(orcid, rec_id)
 
-    Args:
-        config (inspire_utils.config.Config): configuration
-        recid (string): HEP record ID
-        mime_type (string): accept type
 
-    Returns:
-        requests.Response: response form API
-    """
-    server_name = config['SERVER_NAME']
-
-    record_api_endpoint = _get_api_url_for_recid(
-        server_name, config['SEARCH_UI_SEARCH_API'], recid
-    )
-
-    response = requests.get(record_api_endpoint, headers={
-        'Accept': mime_type
-    })
-    response.raise_for_status()
-    return response
-
-
-def _get_hep_record(config, recid):
-    """
+def store_record_in_redis(orcid, rec_id, put_code):
+    """Store the entry <orcid:recid, value> in Redis.
 
     Args:
-        config (inspire_utils.config.Config): configuration
-        recid (string): HEP record ID
+        orcid(string): the author's orcid.
+        rec_id(int): inspire record's id pushed to ORCID.
+        put_code(string): the put_code used to push the record to ORCID.
 
     Returns:
-        dict: HEP record
+        bool: True if the entry is set in Redis, False otherwise.
     """
-    hep_response = _get_record_by_mime(config, recid, 'application/json')
-    return hep_response.json()['metadata']
+    try:
+        with redis_locking_context('orcid_push') as r:
+            orcid_recid_key = get_orcid_recid_key(orcid, rec_id)
+            r.set(orcid_recid_key, put_code)
+            return True
 
+    except RedisLockError:
+        LOGGER.info("Push to ORCID failed for record {}".format(rec_id))
+        return False
 
-def _get_bibtex_record(config, recid):
-    """
 
-    Args:
-        config (inspire_utils.config.Config): configuration
-        recid (string): HEP record ID
+def get_putcode_from_redis(orcid, rec_id):
+    """Retrieve from Redis the put_code for the given ORCID - record id"""
+    try:
+        with redis_locking_context('orcid_push') as r:
+            orcid_recid_key = get_orcid_recid_key(orcid, rec_id)
+            return r.get(orcid_recid_key)
 
-    Returns:
-        dict: BibTeX serialized record
-    """
-    bibtex_response = _get_record_by_mime(config, recid, 'application/x-bibtex')
-    return bibtex_response.text
+    except RedisLockError:
+        LOGGER.info("Push to ORCID failed for record {}".format(rec_id))
```

Comment:
Same as the ones `orcid_push.tasks`, I think that the exception should go up, and be handled by the final caller.